src/Microsoft.Graph.Cli.Core/IO/AuthenticationCacheUtility.cs

```diff
@@ -0,0 +1,56 @@
+using System.Text;
+
+namespace Microsoft.Graph.Cli.Core.IO;
+
+public class AuthenticationCacheUtility : IAuthenticationCacheUtility {
+    private readonly IPathUtility pathUtility;
+
+    private const string AUTHENTICATION_ID_FILE = "authentication-id";
+
+    public AuthenticationCacheUtility(IPathUtility pathUtility)
+    {
+        this.pathUtility = pathUtility;
+    }
+
+    public async Task<(string, string)> ReadAuthenticationIdentifiersAsync(CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        var path = this.GetAuthenticationIdFilePath();
+        if (!File.Exists(path)) {
+            throw new FileNotFoundException();
+        }
+
+        var text = await File.ReadAllTextAsync(path, cancellationToken);
+        if (string.IsNullOrWhiteSpace(text)) throw new AuthenticationIdentifierException("The authentication identifier cache file is empty");
+        var split = text.Split(':', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+        if (split.Length != 2) throw new AuthenticationIdentifierException("The authentication identifier cache file cannot be parsed");
+        
+        return (split[0], split[1]);
+    }
+
+    public async Task SaveAuthenticationIdentifiersAsync(string clientId, string tenantId, CancellationToken cancellationToken = default(CancellationToken)) {
```

Comment:
nit: 
```suggestion
    public async Task SaveAuthenticationIdentifiersAsync(string clientId, string tenantId, CancellationToken cancellationToken = default) {
```

This is the same in the `IAuthenticationCacheUtility.cs`